File: corpus/tier1_library_indirect/ind_hkdf_wrapper.py
import os
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.kdf.hkdf import HKDF


def derive(secret, length=32, info=b"app-context", salt=None):
    kdf = HKDF(
        algorithm=hashes.SHA256(),
        length=length,
        salt=salt or os.urandom(16),
        info=info,
    )
    return kdf.derive(secret)


def derive_pair(secret):
    salt = os.urandom(16)
    enc_key = HKDF(algorithm=hashes.SHA256(), length=32, salt=salt, info=b"enc").derive(secret)
    mac_key = HKDF(algorithm=hashes.SHA256(), length=32, salt=salt, info=b"mac").derive(secret)
    return enc_key, mac_key


if __name__ == "__main__":
    print(derive(b"shared-secret").hex())

File: corpus/tier1_library_indirect/ind_hkdf_wrapper.go
package indirect

import (
	"crypto/sha256"
	"hash"
	"io"

	"golang.org/x/crypto/hkdf"
)

func deriver(secret, salt, info []byte) io.Reader {
	var h func() hash.Hash = sha256.New
	return hkdf.New(h, secret, salt, info)
}

func DeriveKey(secret, salt, info []byte, length int) ([]byte, error) {
	r := deriver(secret, salt, info)
	out := make([]byte, length)
	if _, err := io.ReadFull(r, out); err != nil {
		return nil, err
	}
	return out, nil
}
